intelligent_placer_lib/detection.py

```diff
@@ -1,101 +1,132 @@
-        return None
-
-    p = np.where(source_mask >= 200)
-    mask_points = [[point[1], point[0]] for point in zip(*p)]
-    new_mask_points = cv2.perspectiveTransform(np.float32(mask_points).reshape(-1, 1, 2), matrix)[:, 0, :]
-    for point in new_mask_points:
-        if segmented.shape[0] <= int(point[1]) or int(point[1]) < 0 \
-                or segmented.shape[1] <= int(point[0]) or int(point[0]) < 0:
-            return None
-        if segmented[int(point[1])][int(point[0])] != 0 and segmented[int(point[1])][int(point[0])] != object_class:
-            return None
-        segmented[int(point[1])][int(point[0])] = object_class
-    plt.imshow(segmented + 230)
-    plt.show()
-    print(new_mask_points)
-    return segmented
+    return good_cnts[poly_index], good_cnts[:poly_index]+good_cnts[poly_index+1:]
+
+
+def _get_contour_mask(contour, shape):
+    mask = np.zeros(shape)
+    return cv2.drawContours(mask, [contour], -1, 255, cv2.FILLED)
+
+
+def get_channels_average(object_mask: np.ndarray, object_image: np.ndarray) -> np.ndarray:
+    object_part = np.array([object_image[..., i] * (object_mask / 255).astype('uint8') for i in range(object_image.shape[-1])])
+    return np.sum(np.sum(object_part, axis=1), axis=1) / np.sum(object_mask > 0) / 255
+
 
+def is_red_circle(object_mask: np.ndarray, object_image: np.ndarray, object_contour, circle_contour) -> bool:
+    channels_av = get_channels_average(object_mask, object_image)
+    is_red = not np.isnan(channels_av[0]) and channels_av[0] > 0.6 and channels_av[1] < 0.3 and channels_av[2] < 0.3
+    if not is_red:
+        return False
+    cur_res = cv2.matchShapes(object_contour, circle_contour, cv2.cv2.CONTOURS_MATCH_I3, 0.0)
+    return cur_res < 0.1
 
-def _classify_objects(contours, image: np.ndarray, verbose: bool = False) -> Tuple[np.ndarray, List]:
+
+def is_blue_rectangle(object_mask: np.ndarray, object_image: np.ndarray, object_contour) -> bool:
+    channels_av = get_channels_average(object_mask, object_image)
+    is_blue = not np.isnan(channels_av[2]) and channels_av[2] > 0.6 and channels_av[0] < 0.2 and channels_av[1] < 0.6
+    if not is_blue:
+        return False
+    outer_rect = np.int0(cv2.boxPoints(cv2.minAreaRect(object_contour)))
+    object_area, rect_area = cv2.contourArea(object_contour), cv2.contourArea(outer_rect)
+    return object_area / rect_area > 0.9
+
+
+def _add_segmented_object(segmentation: np.ndarray, contour, object_class: int, verbose: bool = False) -> np.ndarray:
+    cur_segmented = np.zeros(segmentation.shape)
+    cv2.drawContours(cur_segmented, [contour], -1, object_class, cv2.FILLED)
+    cv2.drawContours(segmentation, [contour], -1, object_class, cv2.FILLED)
+    if verbose:
+        plt.imshow(cur_segmented + 220)
+        plt.title(f'segmented class {object_class}')
+        plt.show()
+    return segmentation
+
+
+def _classify_objects(contours, polygon_max_x: int, image: np.ndarray, verbose: bool = False) -> np.ndarray:
```

Comment:
I'm not a good in writing python code, but as for me It is better to use blank lines in order to separate logical blocks of code